checker/src/main/java/org/checkerframework/checker/regex/RegexAnnotatedTypeFactory.java

```diff
@@ -425,8 +422,8 @@ private int getRegexValue(AnnotationMirror anno) {
          * @param anno the annotation to extract the list from
          * @return the extracted list of non-null groups
          */
-        private List<Integer> getEnhancedRegexValue(AnnotationMirror anno) {
-            return AnnotationUtils.getElementValueArray(anno, "value", Integer.class, true);
+        private List<Integer> getEnhancedRegexNonNullGroups(AnnotationMirror anno) {
+            return AnnotationUtils.getElementValueArray(anno, "nonNullGroups", Integer.class, true);
```

Comment:
From what I understand, we have to use `Map.get` instead of iterating over the map. So I overloaded the `AnnotationUtils.getElementValueArray` function to accept an ExecutableElement in place of a CharSequence. The implementation is almost identical. I know we shouldn't duplicate code but we can do away with the CharSequence version when we have removed all of its usages (maybe in a different PR).